src/tools/ceph_objectstore_tool.cc

```diff
@@ -2563,6 +2563,44 @@ int set_size(
   return 0;
 }
 
+int clear_data_digest(ObjectStore *store, coll_t coll, ghobject_t &ghobj, ObjectStore::Sequencer &osr) {
+  if (ghobj.hobj.is_snapdir()) {
```

Comment:
I'm fairly sure that any object like snapdir (not in master anyway) won't have data_digest set, so clear_data_digest() would be a noop.  So I suggest that this check can be removed.